### ISSUE TITLE
JBoss VFS is not supported since 7.12.0

### ISSUE THREAD

**Issue by papegaaij** [nielsbasjes/yauaa#921]
When trying to load the `UserAgentAnalyzer` on WildFly, it fails with the following exception:
```
java.lang.IllegalArgumentException: JBoss VFS is not supported
    at nl.basjes.parse.useragent.utils.springframework.core.io.support.PathMatchingResourcePatternResolver.findPathMatchingResources(PathMatchingResourcePatternResolver.java:480)
    at nl.basjes.parse.useragent.utils.springframework.core.io.support.PathMatchingResourcePatternResolver.getResources(PathMatchingResourcePatternResolver.java:260)
    at nl.basjes.parse.useragent.config.ConfigLoader.findAllResources(ConfigLoader.java:291)
    at nl.basjes.parse.useragent.config.ConfigLoader.loadResources(ConfigLoader.java:192)
    at nl.basjes.parse.useragent.config.ConfigLoader.lambda$load$1(ConfigLoader.java:176)
    at java.util.ArrayList.forEach(ArrayList.java:1511)
    at nl.basjes.parse.useragent.config.ConfigLoader.load(ConfigLoader.java:176)
    at nl.basjes.parse.useragent.AbstractUserAgentAnalyzerDirect$AbstractUserAgentAnalyzerDirectBuilder.build(AbstractUserAgentAnalyzerDirect.java:562)
    at nl.basjes.parse.useragent.AbstractUserAgentAnalyzer$AbstractUserAgentAnalyzerBuilder.build(AbstractUserAgentAnalyzer.java:352)
    at nl.topicus.cobra.app.UserAgentParser.init(UserAgentParser.java:21)
```
This is caused by the manual shading of springframework without keeping the support for JBoss VFS. The code we use to build the `UserAgentAnalyzer` is:
```
UserAgentAnalyzer.newBuilder()
    .hideMatcherLoadStats()
    .withFields(UserAgent.DEVICE_CLASS, UserAgent.AGENT_NAME,
        UserAgent.AGENT_VERSION_MAJOR, UserAgent.OPERATING_SYSTEM_NAME)
    .build();
```

There seems to be no way of loading the resources any other way than via the classpath scan in `ConfigLoader`, which makes it impossible to build the `UserAgentAnalyzer` inside an application running on WildFly (or EAP).

**Comment by nielsbasjes** [comment]
Ah yes ... I honestly thought this was no longer needed.
I'll fix this.

**Comment by nielsbasjes** [comment]
Do you perhaps have (can point me to) a minimalistic reproduction that I can use as an integration test?

**Comment by papegaaij** [comment]
I'm not sure about manually starting a JBoss VFS environment. You could start a war with a ServletLoaderListener or something on WildFly. Would that work for you?

**Comment by nielsbasjes** [comment]
@papegaaij Can you please test for me if version `7.12.3-SNAPSHOT` works for you?

It can be found here:
```
<repositories>
  <repository>
    <id>sonatype-oss-snapshots</id>
    <name>Sonatype OSS Snapshots</name>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    <releases><enabled>false</enabled></releases>
    <snapshots><enabled>true</enabled></snapshots>
  </repository>
</repositories>
```

**Comment by papegaaij** [comment]
@nielsbasjes it was easier for me to build it locally, so I installed the `JBossVFSResourceLoading` branch. I can confirm it works as expected in WildFly now.

**Comment by nielsbasjes** [comment]
Released 7.13.0 with this fix.

**Comment by papegaaij** [comment]
Thanks for the quick fix!
